packages/components/src/navigation/menu-item.js

```diff
@@ -47,14 +50,22 @@ const NavigationMenuItem = ( props ) => {
 		<MenuItemUI className={ classes }>
 			<LinkComponentTag
 				className={ classes }
-				href={ href }
+				href={ ! children.length ? href : null }
 				onClick={ handleClick }
 				{ ...linkProps }
 			>
-				<Text variant="body.small">
-					<span>{ title }</span>
+				<Text
+					className="components-navigation__menu-item-title"
```

Comment:
TIL you could do this. Me and several others have created parent `span`s just to add a class.